pkg/cnab/extensions/solver.go

```diff
@@ -0,0 +1,52 @@
+package extensions
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/deislabs/cnab-go/bundle"
+	"github.com/pkg/errors"
+)
+
+type DependencyLock struct {
+	Name string
+	Tag  string
+}
+
+type DependencySolver struct {
+}
+
+func (s *DependencySolver) ResolveDependencies(bun *bundle.Bundle) ([]DependencyLock, error) {
+	deps, err := ReadDependencies(bun)
+	if deps == nil {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, errors.Wrapf(err, "error executing dependencies for %s", bun.Name)
+	}
+
+	q := make([]DependencyLock, 0, len(deps.Requires))
+	for name, dep := range deps.Requires {
+		bundle := strings.Split(dep.Bundle, ":")[0]
+		version, err := s.ResolveVersion(name, dep)
+		if err != nil {
+			return nil, err
+		}
+		lock := DependencyLock{
+			Name: name,
+			Tag:  fmt.Sprintf("%s:%s", bundle, version),
+		}
+		q = append(q, lock)
+	}
+
+	return q, nil
+}
+
+func (s *DependencySolver) ResolveVersion(name string, dep Dependency) (string, error) {
```

Comment:
Should we handle the case where a dep has no tag (i.e., no `:mytag` value, as in `mycloud/mysql`)?  Would this be valid at this stage?  Or, perhaps it would've already been caught earlier in bundle validation, etc... but maybe as a safe-guard?

Reply:
When no tag is specified, then the resolver is going to pick a version based on the available tags in the registry. I already had an unimplemented error but I see that I missed the case when the tag wasn't specified. I'll update it to print the unimplemented error there too. 

I'm a follow up PR I'm going to implement hitting the registry to resolve all the available tags, cache them, and solve for the best tag based on the constraints (in this case when there's no tag and no range, any tag will do and we can grab the highest version according to semver).